java/build-jar-multiplatform.sh

```diff
@@ -0,0 +1,153 @@
+  echo "Finished building jar for $platform"
+}
+
+copy_jars() {
+  local JAR_DIR="$1"
+  echo "Copy to dir $JAR_DIR"
+  for module in "${RAY_JAVA_MODULES[@]}"; do
+    cp -f "$WORKSPACE_DIR"/java/"$module"/target/*jar "$JAR_DIR"
+  done
+  for module in "${RAY_STREAMING_JAVA_MODULES[@]}"; do
+    cp -f "$WORKSPACE_DIR"/streaming/java/"$module"/target/*jar "$JAR_DIR"
+  done
+  # ray runtime jar and streaming runtime jar are in a dir specifed by maven-jar-plugin
+  cp -f "$WORKSPACE_DIR"/build/java/ray*.jar "$JAR_DIR"
+  cp -f "$WORKSPACE_DIR"/streaming/build/java/streaming*.jar "$JAR_DIR"
+}
+
+# This function assuem all dependencies are installed already.
+build_jars_linux() {
+  build_jars linux
+}
+
+# This function assuem all dependencies are installed already.
+build_jars_darwin() {
+  build_jars darwin
+}
+
+build_jars_multiplatform() {
+  download_jars "ray-runtime-$version.jar" "streaming-runtime-$version.jar"
+  prepare_native
+  build_jars multiplatform false
+}
+
+# Download darwin/windows ray-runtime-$version.jar from s3
+# This function assumes linux jars exist already.
+download_jars() {
+  local wait_time=0
+  local sleep_time_units=60
+
+  for f in "$@"; do
+    for os in 'darwin' 'windows'; do
+      if [[ "$os" == "windows" ]]; then
+        break
+      fi
+      local url="https://ray-wheels.s3-us-west-2.amazonaws.com/jars/$TRAVIS_BRANCH/$TRAVIS_COMMIT/$os/$f"
+      local dest_file="$JAR_BASE_DIR/$os/$f"
+      echo "Jar url: $url"
+      echo "Jar dest_file: $dest_file"
+      while true; do
+        if ! wget -q "$url" -O "$dest_file">/dev/null; then
+          echo "Waiting $url to be ready for $wait_time seconds..."
+          sleep $sleep_time_units
+          wait_time=$((wait_time + sleep_time_units))
+          if [[ wait_time == $((60 * 120)) ]]; then
+            echo "Download $url timeout"
+            exit 1
+          fi
+        else
+          echo "Download $url to $dest_file succeed"
+          break
+        fi
+      done
+    done
+  done
+}
+
+# prepare native binaries and libraries.
+prepare_native() {
+  for os in 'darwin' 'linux'; do
+    cd "$JAR_BASE_DIR/$os"
+    jar xf "ray-runtime-$version.jar" "native/$os"
+    local native_dir="$WORKSPACE_DIR/java/runtime/native_dependencies/native/$os"
+    mkdir -p "$native_dir"
+    rm -rf "$native_dir"
+    mv "native/$os" "$native_dir"
+    jar xf "streaming-runtime-$version.jar" "native/$os"
+    local native_dir="$WORKSPACE_DIR/streaming/java/streaming-runtime/native_dependencies/native/$os"
+    mkdir -p "$native_dir"
+    rm -rf "$native_dir"
+    mv "native/$os" "$native_dir"
+  done
+}
+
+# This function assume all multiplatform binaries are prepared already.
+deploy_jars() {
+  echo "Start deploying jars"
+  cd "$WORKSPACE_DIR"/java
+  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip
+  cd "$WORKSPACE_DIR"/streaming/java
+  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip
+  echo "Finished deploying jars"
+}
+
+case "$1" in
+linux)
```

Comment:
Can you document what each option stands for? 
For example, `linux` means build jars that only contains Linux binaries, and `build_jars_multiplatform` means downloading jars of multiple platforms and packaging them into one jar.